packages/library/src/transaction-confirmation-strategy-timeout.ts

```diff
@@ -0,0 +1,27 @@
+import { Commitment } from '@solana/rpc-core';
+
+type Config = Readonly<{
+    abortSignal: AbortSignal;
+    commitment: Commitment;
+}>;
+
+export async function getTimeoutPromise({ abortSignal: callerAbortSignal, commitment }: Config) {
```

Comment:
Since the `commitment` argument is only used to get the timeout, I'd instead require a `timeout` argument here to make the function more reusable and focus on what it actually needs. Then, we can have a `getTimeoutPromiseFromCommitment` helper method that delegates to that one.

Reply:
Yeah, I did this for ✨reasons✨ that I should have written down, but I was so tired I just posted these PRs and walked away.

First, see this conversation: https://discord.com/channels/428295358100013066/439194979856809985/1164344045842214953

The meta here is that we want to _strongly_ discourage people from using timeouts when confirming transactions. I wouldn't have to do this at all if the `requestAirdrop` RPC method returned the slot until which the transaction is valid. As it is, it only returns a signature, and nothing else.

On the heels of that Discord conversation, I decided to ‘jail’ the timeout inside this helper, and to make the timeout itself contingent on the level of commitment:

* makes it useless for any other purpose
* still lets us change the timeout by pushing a new version

Either way, this is not exported from web3.js, so we can refactor it any time.

So yeah, based on that extra context, should we still change it? I could; the decision of the length of the timeout could be moved to `createSignatureOnlyTransactionConfirmer()` in the next PR, or something, though there are no tests for that function there.

Reply:
Makes sense! I think this function would be helpful in a very generic way like if you want to combine a timeout promise with more promises in parallel for your own app but I don't think this fits inside the scope of web3.js. Thus, I'm happy with "jailing" this helper to specific commitments. 👌